homeassistant/components/honeywell/climate.py

```diff
@@ -488,32 +488,34 @@ async def async_turn_aux_heat_off(self) -> None:
     async def async_update(self) -> None:
         """Get the latest state from the service."""
 
-        async def _async_update(login=False) -> None:
-            """Get the latest state from the service."""
+        async def _login() -> None:
             try:
+                await self._data.client.login()
                 await self._device.refresh()
                 self._attr_available = True
                 self._retry = 0
 
-            except UnauthorizedError:
-                if not login:
-                    try:
-                        await self._data.client.login()
-                    except (
-                        AuthError,
-                        ClientConnectionError,
-                        asyncio.TimeoutError,
-                    ):
-                        self._retry += 1
-                        self._attr_available = self._retry <= RETRY
-                    else:
-                        await _async_update(login=True)
-
-            except (ClientConnectionError, asyncio.TimeoutError):
+            except (
+                AuthError,
```

Comment:
Shouldn't this trigger reauth if still `AuthError` after login?

Reply:
Unfortunately NOT for Honeywell. https://github.com/home-assistant/core/pull/91228  
They throw this error for random periods of time when stuff is happening on the back end.  IF the user reloads the integration or restarts HA and they get this error, it triggers the reauth flow (so they can change it if it's actually different). 
If/when they stop sending 401 errors when the backend is down or rebooting, we could add that back in right here.